logfire/_internal/main.py

```diff
@@ -2032,26 +2036,6 @@ def message(self) -> str:
     def message(self, message: str):
         self._set_attribute(ATTRIBUTES_MESSAGE_KEY, message)
 
-    def end(self, end_time: int | None = None) -> None:
-        """Sets the current time as the span's end time.
-
-        The span's end time is the wall time at which the operation finished.
-
-        Only the first call to this method is recorded, further calls are ignored so you
-        can call this within the span's context manager to end it before the context manager
-        exits.
-        """
-        if self._span is None:  # pragma: no cover
-            raise RuntimeError('Span has not been started')
-        if self._span.is_recording():
-            with handle_internal_errors():
-                if self._added_attributes:
-                    self._span.set_attribute(
-                        ATTRIBUTES_JSON_SCHEMA_KEY, attributes_json_schema(self._json_schema_properties)
-                    )
-
-                self._span.end(end_time)
```

Comment:
We used to only call `self._span.end` here if `self._span` was recording. Is that correct? The changes I made ensure that `end()` always gets called, even on a non-recording span, and that's why the timestamps changed below (because it generates another timestamp in the unconditional call to `end()`)

Reply:
```python
import opentelemetry.trace

import logfire

logfire.configure()

tracer = opentelemetry.trace.get_tracer(__name__)
span = tracer.start_span('foo')
assert span.is_recording()
span.end()
assert not span.is_recording()
span.end()  # Logs a warning: Calling end() on an ended span.
```

So keeping the current behaviour seems best.